{"text": "2023-09-09 07:20:40 | INFO - prelude- {'action': 'Checking the operating environment'}\n", "record": {"elapsed": {"repr": "0:00:00.449930", "seconds": 0.44993}, "exception": null, "extra": {"action": "Checking the operating environment"}, "file": {"name": "claim.py", "path": "/home/runner/work/epic/epic/epic/src/claim.py"}, "function": "prelude", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 25, "message": "prelude", "module": "claim", "name": "__main__", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:40.036888+08:00", "timestamp": 1694215240.036888}}}
{"text": "2023-09-09 07:20:40 | DEBUG - Downloading resource- {'url': 'https://raw.githubusercontent.com/QIN2DIM/hcaptcha-challenger/main/src/objects.yaml', 'to': '/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/models/objects.yaml'}\n", "record": {"elapsed": {"repr": "0:00:00.881332", "seconds": 0.881332}, "exception": null, "extra": {"url": "https://raw.githubusercontent.com/QIN2DIM/hcaptcha-challenger/main/src/objects.yaml", "to": "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/models/objects.yaml"}, "file": {"name": "modelhub.py", "path": "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/modelhub.py"}, "function": "request_resource", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 45, "message": "Downloading resource", "module": "modelhub", "name": "hcaptcha_challenger.onnx.modelhub", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:40.468290+08:00", "timestamp": 1694215240.46829}}}
{"text": "2023-09-09 07:20:41 | INFO - Reloading the local cache of Assets- {'assets_dir': '/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/models/_assets'}\n", "record": {"elapsed": {"repr": "0:00:02.127098", "seconds": 2.127098}, "exception": null, "extra": {"assets_dir": "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/models/_assets"}, "file": {"name": "modelhub.py", "path": "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/modelhub.py"}, "function": "flush_runtime_assets", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 136, "message": "Reloading the local cache of Assets", "module": "modelhub", "name": "hcaptcha_challenger.onnx.modelhub", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:41.714056+08:00", "timestamp": 1694215241.714056}}}
{"text": "2023-09-09 07:20:41 | DEBUG - Pulling Assets index file- {'url': 'https://api.github.com/repos/QIN2DIM/hcaptcha-challenger/releases'}\n", "record": {"elapsed": {"repr": "0:00:02.128254", "seconds": 2.128254}, "exception": null, "extra": {"url": "https://api.github.com/repos/QIN2DIM/hcaptcha-challenger/releases"}, "file": {"name": "modelhub.py", "path": "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/modelhub.py"}, "function": "flush_runtime_assets", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 150, "message": "Pulling Assets index file", "module": "modelhub", "name": "hcaptcha_challenger.onnx.modelhub", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:41.715212+08:00", "timestamp": 1694215241.715212}}}
{"text": "2023-09-09 07:20:44 | INFO - execute task- {'name': 'claim_epic_games'}\n", "record": {"elapsed": {"repr": "0:00:05.037793", "seconds": 5.037793}, "exception": null, "extra": {"name": "claim_epic_games"}, "file": {"name": "models.py", "path": "/home/runner/work/epic/epic/epic/src/services/models.py"}, "function": "execute", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 105, "message": "execute task", "module": "models", "name": "services.models", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:44.624751+08:00", "timestamp": 1694215244.624751}}}
{"text": "2023-09-09 07:20:44 | INFO - claim_epic_games- {'action': 'Try to flush cookie'}\n", "record": {"elapsed": {"repr": "0:00:05.091868", "seconds": 5.091868}, "exception": null, "extra": {"action": "Try to flush cookie"}, "file": {"name": "claim.py", "path": "/home/runner/work/epic/epic/epic/src/claim.py"}, "function": "claim_epic_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 57, "message": "claim_epic_games", "module": "claim", "name": "__main__", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:44.678826+08:00", "timestamp": 1694215244.678826}}}
{"text": "2023-09-09 07:20:46 | SUCCESS - login- {'result': 'token has not expired'}\n", "record": {"elapsed": {"repr": "0:00:06.629345", "seconds": 6.629345}, "exception": null, "extra": {"result": "token has not expired"}, "file": {"name": "epic_games.py", "path": "/home/runner/work/epic/epic/epic/src/services/agents/epic_games.py"}, "function": "_login", "level": {"icon": "‚úÖ", "name": "SUCCESS", "no": 25}, "line": 129, "message": "login", "module": "epic_games", "name": "services.agents.epic_games", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:46.216303+08:00", "timestamp": 1694215246.216303}}}
{"text": "2023-09-09 07:20:57 | SUCCESS - flush_token- {'path': PosixPath('/home/runner/work/epic/epic/epic/user_data_dir/epic-games@epicgames/ctx_cookie.json')}\n", "record": {"elapsed": {"repr": "0:00:18.308634", "seconds": 18.308634}, "exception": null, "extra": {"path": "/home/runner/work/epic/epic/epic/user_data_dir/epic-games@epicgames/ctx_cookie.json"}, "file": {"name": "epic_games.py", "path": "/home/runner/work/epic/epic/epic/src/services/agents/epic_games.py"}, "function": "flush_token", "level": {"icon": "‚úÖ", "name": "SUCCESS", "no": 25}, "line": 150, "message": "flush_token", "module": "epic_games", "name": "services.agents.epic_games", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:57.895592+08:00", "timestamp": 1694215257.895592}}}
{"text": "2023-09-09 07:20:58 | WARNING - Failed to get order history, cookie may have expired- {}\n", "record": {"elapsed": {"repr": "0:00:18.914399", "seconds": 18.914399}, "exception": null, "extra": {}, "file": {"name": "epic_games.py", "path": "/home/runner/work/epic/epic/epic/src/services/agents/epic_games.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 285, "message": "Failed to get order history, cookie may have expired", "module": "epic_games", "name": "services.agents.epic_games", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:58.501357+08:00", "timestamp": 1694215258.501357}}}
{"text": "2023-09-09 07:20:58 | INFO - claim_weekly_games- {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/spelldrifter'}\n", "record": {"elapsed": {"repr": "0:00:19.298215", "seconds": 19.298215}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/spelldrifter"}, "file": {"name": "epic_games.py", "path": "/home/runner/work/epic/epic/epic/src/services/agents/epic_games.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 162, "message": "claim_weekly_games", "module": "epic_games", "name": "services.agents.epic_games", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:20:58.885173+08:00", "timestamp": 1694215258.885173}}}
{"text": "2023-09-09 07:21:09 | ERROR - An error has been caught in function '_run', process 'MainProcess' (3499), thread 'MainThread' (140115130833984):- {}\nTraceback (most recent call last):\n  File \"/home/runner/work/epic/epic/epic/src/claim.py\", line 94, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n> File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n  File \"/home/runner/work/epic/epic/epic/src/claim.py\", line 90, in run\n    await agent.execute(sequence=[claim_epic_games], headless=True)\n  File \"/home/runner/work/epic/epic/epic/src/services/models.py\", line 113, in execute\n    await container(context)\n  File \"/home/runner/work/epic/epic/epic/src/claim.py\", line 79, in claim_epic_games\n    await epic.claim_weekly_games(page, promotions)\n  File \"/home/runner/work/epic/epic/epic/src/services/agents/epic_games.py\", line 176, in claim_weekly_games\n    await expect(cta_btn).to_have_text(\"View In Cart\")\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 19964, in to_have_text\n    await self._impl_obj.to_have_text(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 455, in to_have_text\n    await self._expect_impl(\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 70, in _expect_impl\n    raise AssertionError(f\"{out_message}\\nActual value: {actual} {log}\")\nAssertionError: Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\nwaiting for locator(\"xpath=//aside//button[@data-testid='add-to-cart-cta-button']\")\n", "record": {"elapsed": {"repr": "0:00:29.530570", "seconds": 29.53057}, "exception": {"type": "AssertionError", "value": "Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\nwaiting for locator(\"xpath=//aside//button[@data-testid='add-to-cart-cta-button']\")", "traceback": true}, "extra": {}, "file": {"name": "events.py", "path": "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/events.py"}, "function": "_run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 80, "message": "An error has been caught in function '_run', process 'MainProcess' (3499), thread 'MainThread' (140115130833984):", "module": "events", "name": "asyncio.events", "process": {"id": 3499, "name": "MainProcess"}, "thread": {"id": 140115130833984, "name": "MainThread"}, "time": {"repr": "2023-09-09 07:21:09.117528+08:00", "timestamp": 1694215269.117528}}}
